src/main/java/edu/wpi/grip/ui/PaletteView.java

```diff
@@ -49,6 +62,12 @@ public void clearOperations() {
 
     @Subscribe
     public void onOperationAdded(OperationAddedEvent event) {
-        this.operations.getChildren().add(new OperationView(this.eventBus, event.getOperation()));
+        OperationView view = new OperationView(this.eventBus, event.getOperation());
+        view.determineVisibility(this.operationSearch.getText());
```

Comment:
Have a listener specifically for filtering operations:

    final InvalidationListener filterOperations = observable ->
        this.operations.getChildren().forEach(node -> {
            final Operation operation = ((OperationView) node).getOperation();
            node.setVisible(myTextSearchAlgorithmMatches(this.textField.getText(), operation.getName()));
        });

Then make sure it's called at the appropriate times using JavaFX listeners

    // Update the visibility of each node when either the list of nodes or the filter text changes
    this.getChildren().addListener(filterOperations);
    this.textField.textProperty().addListener(filterOperations);